azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -0,0 +1,175 @@
+            raise PolicyError(msg)
+
+    @classmethod
+    def log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_event=True):
+        """
+        Log information to console and telemetry.
+        """
+        if is_success:
+            logger.info(msg)
+        else:
+            logger.error(msg)
+        if send_event:
+            add_event(op=op, message=msg, is_success=is_success)
+
+    @staticmethod
+    def is_policy_enforcement_enabled():
+        """Check whether user has opted into policy enforcement feature"""
+        # TODO: The conf flag will be removed post private preview. Before public preview, add checks
+        # according to the planned user experience (TBD).
+        return conf.get_extension_policy_enabled()
+
+    @staticmethod
+    def check_policy_enforcement_supported():
+        """
+        Check that both platform architecture and distro/version are supported.
+        If supported, do nothing.
+        If not supported, raise PolicyError with user-friendly error message.
+        """
+        osutil = get_osutil()
+        arch = osutil.get_vm_arch()
+        # TODO: surface as a user error with clear instructions for fixing
+        msg = "Attempted to enable policy enforcement, but feature is not supported on "
+        if arch not in POLICY_SUPPORTED_ARCHITECTURE:
+            msg += " architecture " + str(arch)
+        elif DISTRO_NAME not in POLICY_SUPPORTED_DISTROS_MIN_VERSIONS:
+            msg += " distro " + str(DISTRO_NAME)
+        else:
+            min_version = POLICY_SUPPORTED_DISTROS_MIN_VERSIONS.get(DISTRO_NAME)
+            if DISTRO_VERSION < min_version:
+                msg += " distro " + DISTRO_NAME + " " + DISTRO_VERSION + ". Policy is only supported on version " + \
+                        str(min_version) + " and above."
+            else:
+                return  # do nothing if platform is supported
+        raise PolicyError(msg)
+
+
+    @property
+    def policy_engine_enabled(self):
+        """This property tracks whether the feature is enabled and Regorus engine has been successfully initialized"""
+        return self._policy_engine_enabled
+
+    def evaluate_query(self, input_to_check, query):
+        """
+        Input_to_check is the input we want to check against the policy engine (ex: extensions we want to install).
+        Input_to_check should be a dict. Expected format:
+        {
+            "extensions": {
+                "<extension_name_1>": {
+                    "signingInfo": {
+                        "extensionSigned": <true, false>
+                    }
+                }, ...
+        }
+
+        The query parameter specifies the value we want to retrieve from the policy engine.
+        Example format for query: "data.agent_extension_policy.extensions_to_download"
+        """
+        # This method should never be called if policy is not enabled, this would be a developer error.
+        if not self.policy_engine_enabled:
```

Comment:
```suggestion
        if not self._policy_engine_enabled:
```
or 
```suggestion
        if not self.policy_engine_enabled():
```

Reply:
Whoa! great catch @maddieford! 

pylint disappoints me more often than I'd like. But if the linter fails, the next step would be unit tests. @mgunnala - would you review unit tests and see why they missed this basic scenario? thanks!

Reply:
@maddieford not sure I'm understanding the issue - this came from a previous comment to make policy_engine_enabled( ) into a @property method. My understanding is we can just access "self.policy_engine_enabled" since it's acting as a property. Is this not the case?

Reply:
@mgunnala - the parenthesis are missing from the function call. As written currently, that condition is always True

maddieford points out that you should add the parenthesis or use the private property

Reply:
This is a bad bug, easy to miss. The linter (nor the unit tests) didn't point it out.

Reply:
ah, it's a property!

Reply:
I thought @property methods in python didn't require the parentheses.

I have a unit test looking for exactly this scenario - "test_eval_query_should_throw_error_when_disabled()". It passes, and when I step through it, the function is being called as expected.

Reply:
yes, i think we misread the code

Reply:
but, related to this property.. is this needed at all? see my other comment. looks like a left-over

Reply:
Replied to your other comment